core/engine/src/vm/opcode/args.rs

```diff
@@ -807,3 +939,654 @@ impl Argument for (VaryingOperand, ThinVec<u32>) {
         ((first.into(), rest), pos + 6 + total_len * 4)
     }
 }
+
+#[cfg(test)]
+mod test {
+
+  use std::vec;
+  use thin_vec::thin_vec;
+  use super::*;
+
+  fn test_write_arg<T: Argument>(test_data: T) -> Vec<u8> {
+    let mut v: Vec<u8> = Vec::new();
+    test_data.encode(&mut v);
+    return v
+  }
+
+  // This can fail depending on Argument/opcode combination given (too short)
+  fn test_read_arg<T: Argument>(opcode: Box<[u8]>, pos: usize) -> (T, usize) {
+    let r: &[u8] = &opcode;
```

Comment:
```suggestion
  fn test_read_arg<T: Argument, B: AsRef<[u8]>>(bytecode: B, pos: usize) -> (T, usize) {
    let r: &[u8] = bytecode.as_ref();
```

We can add a generic type that can be turned into a reference so we can give an array directly.

```rust
    test_read_arg::<VaryingOperand>([0,0xFF], 0);
```

See: https://doc.rust-lang.org/std/primitive.array.html#impl-AsRef%3C%5BT%5D%3E-for-%5BT;+N%5D
See: https://doc.rust-lang.org/std/convert/trait.AsRef.html